src/lib.rs

```diff
@@ -348,6 +352,92 @@ pub(crate) trait Castable {
     type RustType;
 }
 
+/// Defines a new [`Castable`] opaque struct with [`OwnershipBox`] ownership.
+///
+/// Expects to be invoked with a visibility specifier, the struct keyword, a struct name, and
+/// in parens, the Rust type that the opaque struct wraps. Similar to defining a normal struct
+/// that wraps a single type as its `.0` member.
+macro_rules! box_castable {
+    (
+        $(#[$comment:meta])*
+        $enum_vis:vis struct $name:ident($rust_type:ty);
+    ) => {
+        crate::castable!(OwnershipBox $(#[$comment])* $enum_vis $name $rust_type);
+    };
+}
+
+pub(crate) use box_castable;
+
+/// Defines a new [`Castable`] opaque struct with [`OwnershipArc`] ownership.
+///
+/// Expects to be invoked with a visibility specifier, the struct keyword, a struct name, and
+/// in parens, the Rust type that the opaque struct wraps. Similar to defining a normal struct
+/// that wraps a single type as its `.0` member.
+macro_rules! arc_castable {
+    (
+        $(#[$comment:meta])*
+        $enum_vis:vis struct $name:ident($rust_type:ty);
```

Comment:
Calling this `$enum_vis` is confusing because we're always working with `struct`. But probably we can do away with this variable since it will always be `pub`, match on `pub struct`, and emit `pub struct`.

Reply:
Oops, that was copy-pasta from borrowing from the Rustls enum builder macro.

I've taken your suggestion to remove the parameter and just use `pub` throughout.